File: OP_PROJECT_PYTHON/src/game/Plants/Guarana.py
from src.game.Plant import Plant


class Guarana(Plant):
    GUARANA_STRENGTH = 0
    GUARANA_TYPE = "Guarana"

    def __init__(self, position, world):
        super().__init__(Guarana.GUARANA_TYPE, Guarana.GUARANA_STRENGTH, position, world)

    def check_power_up(self, attacker):
        attacker.set_strength(attacker.get_strength() + 3)
        info = "Guarana increased strength of " + attacker.get_type()
        self.world.display_information(info)

File: OP_PROJECT_JAVA/src/game/Plants/Guarana.java
package game.Plants;

import game.Organism;
import game.Plant;
import game.Position;
import game.World;
public class Guarana extends Plant {

    private static final int GUARANA_STRENGTH = 0;
    private static final String GUARANA_TYPE = "Guarana";

    public Guarana(Position position, World world) {
        super(GUARANA_TYPE, GUARANA_STRENGTH, position, world);
    }

    @Override
    protected void checkPowerUp(Organism attacker) {
        attacker.setStrength(getStrength()+3);
        String info = "Guarana increased strength of " + attacker.getType();
        world.displayInformation(info);
    }
}
